eng/doc/DeveloperGuide.md

```diff
@@ -138,6 +138,66 @@ You can use your build of `go` in VS Code by following these steps:
 1. Open the command palette.
 1. Search for `Developer: Reload Window` and select it.
 
-## Making changes to go/src
+## Making Changes to `go/src`
 
-TODO
\ No newline at end of file
+### Preparing `go/src` folder
+As mentioned above, after downloading the submodule using the command:
+
+```bash
+git submodule update --init --recursive
+```
+
+and applying all changes from patch files with:
+
+```bash
+git go-patch apply
+```
+
+fully prepared go/src directory will be created. From this point forward, all tasks related to vendoring, updating the go.mod file, and running tests should exclusively utilize the Go toolchain provided by the go submodule.
+
+It is crucial to note that the Go version and source code within the submodule may differ from your local setup. As a result, tests that pass in your local environment may fail in the submodule environment, and tests that pass in the submodule may fail locally. Always validate your changes within the submodule context to ensure compatibility and consistency.
+
+### Making Changes
+
+Once the `go/src` folder is prepared, any modifications made to this directory will be tracked by the Git history of the submodule. You can view these changes by running:
+
+```bash
+git status
+```
+
+To create patch files from the changes, you must commit them. Only committed changes will be extracted by go-patch and included in the patch files.
+
+### Generating New Patch Files
+
+After making changes in the `go/src` directory, you must commit your changes following the standard Git process. For example:
+
+```bash
+git add . --all
+git commit -m "example"
+```
+
+This will create a commit with the message "example" in the Git log. 
+
+Then, when you run:
+
+```bash
+git go-patch extract
+```
+
+The `go-patch` command will generate a patch file under the `go/patches` directory. The patch file will be prefixed with a serial number (one greater than the number of existing patch files), followed by a dash-separated commit message.
+
+### Squashing Changes to Existing Patch Files
+
+Creating new patch files is not always necessary when there are existing patch files with similar purposes for the same files. In such cases, you can squash new commits on top of the existing ones to update their contents. The go-patch extract command will detect the differences in these commits and regenerate the patch files with the updated contents.
+
+Before starting work, please check the go/patches folder for any existing patch files related to the files you're working on. This helps maintain a clean repository by avoiding redundant patch files.
+
+### Rolling Back Changes in `go/src`
```

Comment:
The concept explain in this section is important, but I don't think it deserved it's won section. What if you add a new paragraph to the `Making Changes` section saying something like this:

> Before starting any new work in `go/src`, make sure that the Go submodule has all the patches are applied and that it doesn't contain previous changes. You can do that by running the following command:

```
git go-patch apply -f
```

Reply:
This paragraph is not related to whether patches have already been applied. It states that before starting to work, one must check if a new patch file needs to be created or if the changes can be added to an existing patch file.

Reply:
Srry, my comment was referring to the `### Rolling Back Changes in go/src` section, I put the comment in the wrong line 😓 .

Reply:
I think it makes its own sense, since the storyline in here goes as;
Beginning
Process
Final.
So final cleanup can require more additional steps in future. and IMO keeping it in this way is making much more sense